### ISSUE TITLE
Blocky segmentation masks

### ISSUE THREAD

**Issue by aleksmirosh** [NASA-IMPACT/hls-foundation-os#55]
Hello, thank you for your incredible work. This model shows great IoU on my data, much better than all I tested before.
But even after finetune masks look blocky. Like here, like small squares:
<img width="470" alt="mask" src="https://github.com/NASA-IMPACT/hls-foundation-os/assets/74064180/963e571b-c676-43a9-9a41-9474f54a2c88">

Do you have some suggestions please how I can deal with it?

**Comment by aleksmirosh** [comment]
any suggestions?

**Comment by HamedAlemo** [comment]
@aleksmirosh we had another user trying this and having similar pattern in their prediction. The issue for them turned out to be that their input data was between [0, 1] but you need to multiply them by 10000 to be integers. 
cc @paolofraccaro

**Comment by aleksmirosh** [comment]
thank you @HamedAlemo for your response, I much appreciate it.
I tried the model on satellite but not Sentinal data, it is already between 0 and 60000.
I fond that this block is 16x16 size. Is this the problem with the encoder/decoder? Should I finetune it?
I much appreciate your help with the issue

**Comment by HamedAlemo** [comment]
I think @thesujitroy or @paolofraccaro should be able to help with this.

**Comment by CarlosGomes98** [comment]
Hi @aleksmirosh . This is somewhat confusing to me, as the predictions should be either 0 or 1 for each pixel, so completely dark or completely white, So I suspect at least some of this is due to distortion being introduced by compression in your image (are you saving it as JPEG perhaps?)

Its possible the output of the image is also being affected by the 16x16 patch size though, as you observed. You could try using a smaller patch size on your data, although it may take some more training epochs.

**Comment by aleksmirosh** [comment]
Hi @CarlosGomes98  thank you for the response. I much appreciate your time.
My predictions are probabilities, I tested the original finetune script on the provided sentinel floods, and predictions were probabilities between 0 and 1 too. I attached mask I got after original finetune process on flood data.
<img width="302" alt="mask_flood" src="https://github.com/NASA-IMPACT/hls-foundation-os/assets/74064180/e705be82-9d6d-4232-8dcf-73941b268fe6">

For to my custom finetune the input is TIFF and I save it as TIFF.
I tried different constants and normalizations, but the results are the same. 

I will try to experiment with patch size. Does it make sense to try a bigger patch size? In this case, I must use batch size = 1, just memory limit.
